amplify/backend/function/getKnowledgeGraph/src/index.js

```diff
@@ -1,20 +1,47 @@
 /**
  * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
  */
+
+const getPapersByTitle = require("/opt/utils/getPapersByTitle");
+const querySemanticScholarByPaperId = require("/opt/utils/querySemanticScholarByPaperId");
+
+async function getReturnMessages({ messageContent }) {
+  const headers = {
+    "Access-Control-Allow-Origin": "*",
+    "Access-Control-Allow-Headers": "*",
+  };
+  const returnMessage = {
+    statusCode: 200,
+    headers,
+    body: JSON.stringify(messageContent),
+  };
+  return returnMessage;
+}
 exports.handler = async (event) => {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const axios = require("axios");
-  return await axios
-    .get(`https://api.semanticscholar.org/graph/v1/paper/` + event.paperId, {
-      params: {
-        fields:
-          "authors,citations,citationCount,references,referenceCount,title",
-      },
-      headers: {
-        "x-api-key": process.env.SC_API_KEY,
-      },
-    })
-    .then(function (paperInfo) {
+  console.log(`EVENT: ${JSON.stringify(event)}`);
+  const input = JSON.parse(event.body);
+
+  let paperId = input.paperId;
+  let paperTitle = input.paperTitle;
+
+  try {
+    if (paperId === null && paperTitle !== null) {
+      const paperDetails = await getPapersByTitle({
+        paperTitle,
+        fieldsToGet: "paperId,title,authors",
+        limit: 1,
+      });
+      console.log("Paper details", paperDetails);
+      if (paperDetails.data.length > 0) paperId = paperDetails.data[0].paperId;
+    }
+    if (paperId !== null) {
+      const paperInfo = await querySemanticScholarByPaperId({
+        paperId,
+        fieldsToGet:
+          "title,authors,references,references.authors,citations,citations.authors,referenceCount,citationCount",
```

Comment:
Could you please add `references.title` and `citations.title` to the fields? I don't know why I didn't add them myself, but when testing the component, I see they are missing from the response.

Reply:
done! Test it out and let me know